src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -285,6 +285,16 @@ def __init__(
         if isinstance(image_processor, list):
             image_processor = ComposeProcessing(image_processor)
 
+        has_image_permute = False
+
+        for image_processing in image_processor.processings:
+            if isinstance(image_processing, ImagePermute):
+                has_image_permute = True
+                break
```

Comment:
What do you think about this ?
```python
has_image_permute  = any(isinstance(image_processing, ImagePermute) for image_processing in image_processor.processings)
```

Reply:
Agree, looks way better. Fixed in a4632b8c73d28b37f1fd52ca3fa321370d83b912